src/fs/fs.ts

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) 2023, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import * as nodeFs from 'node:fs';
+// yes, we're going to import it even though it might not be used.
+// the alternatives were all worse without top-level await (iife, runtime errors from something trying to use it before it's initialized)
+import * as memfs from 'memfs';
+import type { VirtualFs } from './types';
+
+export let fs: VirtualFs;
+
+const isWeb = (): boolean => process.env.FORCE_MEMFS === 'true' || 'window' in globalThis || 'self' in globalThis;
```

Comment:
Why not use `Global.isWeb`?  E.g.,

`const isWeb = (): boolean => process.env.FORCE_MEMFS === 'true' || Global.isWeb`

Reply:
Ahh, because circular ref.  Got it.